Create submission consent screen (QR code flow) (EXPOSUREAPP-3676)

chiljamgossow opened the pull request:
Creates new screen for consent to submission of keys in QR code flow
Removes old info screen and dialog
Updates navigation
Link to data privacy screen
Move legal text

Still missing: correct icons -> moved to new ticket 3972

![3676-submission-consent-screen](https://user-images.githubusercontent.com/49635654/100111301-3ecf8600-2e6e-11eb-808d-eb4ff2f17d18.png)

chiljamgossow committed:
create fragment, viewmodel, layout

chiljamgossow committed:
Merge branch 'release/1.9.x' into feature/3676-submission-consent-screen

chiljamgossow committed:
create fragment, viewmodel, layout

chiljamgossow committed:
remove info fragment

chiljamgossow committed:
clean up

chiljamgossow committed:
clean up

chiljamgossow committed:
clean up

chiljamgossow committed:
clean up

chiljamgossow committed:
clean up

chiljamgossow committed:
Merge branch 'feature/3676-submission-consent-screen-old' into feature/3731-submission-consent-screen

# Conflicts:
#	Corona-Warn-App/src/main/res/layout/fragment_submission_qr_code_info.xml
#	Corona-Warn-App/src/main/res/values-bg/strings.xml
#	Corona-Warn-App/src/main/res/values-en/strings.xml
#	Corona-Warn-App/src/main/res/values-pl/strings.xml
#	Corona-Warn-App/src/main/res/values-ro/strings.xml
#	Corona-Warn-App/src/main/res/values-tr/strings.xml
#	Corona-Warn-App/src/main/res/values/strings.xml

chiljamgossow committed:
clean up

chiljamgossow committed:
Merge branch 'release/1.9.x' into feature/3676-create-submission-consent-screen

chiljamgossow committed:
navigation change

janetback commented:
Hi @chiljamgossow - I am just wondering why this PR includes files for languages other than the source language German? These are normally update by deliveries from translation.

chiljamgossow committed:
navigation change

chiljamgossow commented:
> Hi @chiljamgossow - I am just wondering why this PR includes files for languages other than the source language German? These are normally update by deliveries from translation.

Hi @janetback I deleted a screen and also removed the strings that are now unused.

vaubaehn commented:
For my taste looks nice in general.

@janetback and @SabineLoss : I would suggest to move the text "Die App kann nicht gleichzeitg mehrere Tests verwalten" up one icon/bullet point, directly following "Scannen Sie nur Ihren eigenen Test". (There should also be a "." behind "Test" for consistency.) With regard to content it maybe fits better there.
I also would suggest to change the wording of "Ihr Test kann nur einmal gescannt werden." to "Ihr Test kann nur ein einziges Mal gescannt werden." to emphasize even more the loss of the test retrieval after deleting the test from CWA.
What do you think?

janetback commented:
Hi @vaubaehn, thanks for your input.
We (UA) have looked at your comments and have decided to make the following changes. We didn't think it was necessary to add 'ein einziges Mal', but hope that our improvements will make things clearer for the user.

•	Scannen Sie nur Ihren eigenen Test.
•	Sie können jeden Test nur einmal scannen. Wenn Sie einen gescannten Test löschen, können Sie ihn nicht
        erneut scannen.
•	Die App kann nur einen Test verwalten. Wenn Ihnen ein aktuellerer Test vorliegt, löschen Sie zunächst 
        den vorhandenen Test. Danach können Sie den aktuellen Test in der App registrieren. 

@chiljamgossow - I will enter the new text proposal for the bulleted list into (EXPOSUREAPP-3676).

kolyaopahle commented on Corona-Warn-App/src/main/res/layout/fragment_submission_consent.xml:
```diff
@@ -0,0 +1,144 @@
+            style="@style/buttonPrimary"
+            android:layout_width="@dimen/match_constraint"
+            android:layout_height="wrap_content"
+            android:text="@string/submission_positive_other_warning_button"
```
maybe rename the string resources so that they represent the usage place more accurately

chiljamgossow replied:
renamed to submission_accept_button

kolyaopahle commented on Corona-Warn-App/src/main/res/layout/fragment_submission_consent.xml:
```diff
@@ -0,0 +1,144 @@
+            android:layout_height="wrap_content"
+            android:text="@string/submission_positive_other_warning_button"
+            android:textAllCaps="true"
+            android:onClick="@{ () -> viewModel.onConsentButtonClick()}"
```
could we not inject the viewModel and instead just set the listener in fragment code?

chiljamgossow replied:
This is possible. I decided to use data binding to kkep the fragment clean

kolyaopahle commented on Corona-Warn-App/src/test/java/de/rki/coronawarnapp/ui/submission/qrcode/consent/SubmissionConsentViewModelTest.kt:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    @Test
+    fun testBackPressButton() {
```
is this actually the back button being tested?

chiljamgossow replied:
Thanks for spotting this. I changed this.

kolyaopahle reviewed (commented):
Looks nice as far as i can see, some very minor nitpicks 👍

janetback commented:
Hi @vaubaehn, thanks for your feedback. We will add the missing '.' and take your suggestions into account for future improvements.

@chiljamgossow - as discussed, could you please add the missing punctuation? Thanks.

chiljamgossow commented:
> @chiljamgossow - as discussed, could you please add the missing punctuation? Thanks.

done

chiljamgossow committed:
navigate to data privacy
review comments

chiljamgossow committed:
Merge branch 'release/1.9.x' into feature/3676-create-submission-consent-screen

chiljamgossow committed:
move text to legal_strings.xml
test code

chiljamgossow committed:
legal_strings.xml

janetback reviewed (dismissed):
UA approved

BMItter commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/qrcode/consent/SubmissionConsentFragment.kt:
```diff
@@ -0,0 +1,47 @@
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.viewModel = viewModel
+        binding.lifecycleOwner = this
```
_binding.lifecycleOwner = this_ should be replaced with this.viewLifecycleOwner 
(can lead to fragment issues e.g. while navigating)

It's also little bit redundant because you already use viewBindingLazy above.
So to avoid this bug either adjust the lifecycleOwner line or simply remove it :snowman:

chiljamgossow replied:
I removed it. I missed that it is already applied.

chiljamgossow committed:
remove lifecycle assignment as it is redundant

BMItr reviewed (approved):
code lgtm

vaubaehn commented:
Hi @janetback , thanks for your kind reply. From the user's perspective, I also liked your approach from the later erased comment for its clarity, but full understanding that it's difficult to find the balance between wording and length, without breaking the layouts :wink: Thanks for your important contributions to this big project :+1: -> this goes also to @SabineLoss

janetback commented:
Hi @vaubaehn , yes, we are definitely keeping the proposal that I then erased - we are still hoping to be able to implement it at a slightly later date. 
cc @SabineLoss

janetback reviewed (approved):
UA reviewed
